src/utils/componentSwapData.ts

```diff
@@ -12,3 +18,166 @@ export interface ComponentSwapData {
+    '0x39AA39c021dfbaE8faC545936693aC917d5E7563',
+    6,
+    provider
+  )
+  const cUsdtExchangeRate = await getExchangeRate(
+    '0xf650C3d88D12dB855b8bf7D11Be6C55A4e07dCC9',
+    6,
+    provider
+  )
+
+  const requiredDai = issuanceUnits[0].mul(cDaiExchangeRate)
+  const requiredUsdc = issuanceUnits[1].mul(cUsdcExchangeRate)
+  // const requiredUsdt = issuanceUnits[2].mul(cUsdtExchangeRate)
+
+  console.log(requiredDai.toString())
+  console.log(requiredUsdc.toString())
+  // console.log(requiredUsdt.toString())
+
+  const buyUnderlyingAmountDai =
+    indexTokenMix !== IndexTokenMix.UNWRAPPED_ONLY
+      ? requiredDai
+      : issuanceUnits[0]
+  // cUSDC is only used in test case WRAPPED_ONLY
+  const buyUnderlyingAmountUsdc =
+    indexTokenMix === IndexTokenMix.WRAPPED_ONLY
+      ? requiredUsdc
+      : issuanceUnits[1]
+  // const buyUnderlyingAmountUsdt =
+  //   indexTokenMix !== IndexTokenMix.UNWRAPPED_ONLY
+  //     ? requiredUsdt
+  //     : issuanceUnits[2]
+
+  return [
+    {
+      underlyingERC20: dai,
+      buyUnderlyingAmount: buyUnderlyingAmountDai,
+      dexData: getStaticIssuanceSwapData(inputToken, dai),
+    },
+    {
+      underlyingERC20: usdc,
+      buyUnderlyingAmount: buyUnderlyingAmountUsdc,
+      dexData: getStaticIssuanceSwapData(inputToken, usdc),
+    },
+    {
+      underlyingERC20: usdt,
+      buyUnderlyingAmount: BigNumber.from(0),
+      dexData: getStaticIssuanceSwapData(inputToken, usdt),
+    },
+  ]
+}
+
+export function getRedemptionComponentSwapData(
+  outputToken: string
+): ComponentSwapData[] {
+  return [
+    {
+      underlyingERC20: dai,
+      buyUnderlyingAmount: BigNumber.from(0), // not used in redeem
+      dexData: getStaticRedemptionSwapData(dai, outputToken),
+    },
+    {
+      underlyingERC20: usdc,
+      buyUnderlyingAmount: BigNumber.from(0), // not used in redeem
+      dexData: getStaticRedemptionSwapData(usdc, outputToken),
+    },
+    {
+      underlyingERC20: usdt,
+      buyUnderlyingAmount: BigNumber.from(0), // not used in redeem
+      dexData: getStaticRedemptionSwapData(usdt, outputToken),
+    },
+  ]
+}
+
+async function getExchangeRate(
+  cErc20: string,
+  decimals: number,
+  provider: JsonRpcProvider
+): Promise<BigNumber> {
+  const contract = new Contract(cErc20, CErc20Abi, provider)
+  // Returns the current exchange rate as an unsigned integer, scaled by 1 * 10^(18 - 8 + Underlying Token Decimals).
+  // https://docs.compound.finance/v2/ctokens/#exchange-rate
+  const exchangeRate = await contract.callStatic.exchangeRateCurrent()
+  const scale = Math.pow(10, 18 - 8 + decimals)
+  const divByScale = Math.pow(10, decimals - 8)
+  console.log(scale, divByScale)
+  console.log(exchangeRate.toString(), '/////// exchange rate scaled')
+  const realExchangeRate =
+    scale > 1e18 ? exchangeRate.div(divByScale) : exchangeRate
+  console.log(realExchangeRate.toString(), '/////// exchange rate')
+  return realExchangeRate
```

Comment:
Please make sure I got this right from the tests: https://github.com/IndexCoop/index-coop-smart-contracts/blob/master/test/exchangeIssuance/flashMintWrapped.spec.ts#L392

Reply:
Actually, right now this wouldn't be used. So will probably remove if it won't.